### ISSUE TITLE
Better info for PYTHONHOME, PYTHONPATH, LD_LIBRARY_PATH, DYLD_LIBRARY_PATH, other path vars

### ISSUE THREAD

**Issue by jakirkham** [conda/conda#7208]
It's really surprising (or maybe not surprising for those of us familiar at this stage) how many problems are caused by users setting these environment variables and getting "surprising" behavior from `conda` environments. Though these often get discovered at the last moment in a user's issue report, which is bad for time usage and experience on both ends. Would be good if we could catch these earlier and/or avoid them entirely. Here are some rough ideas (some are good, some are scary).

* Listing any of these vars set in `conda info`.
* Warning users if these are already set when activating an environment.
* Warning after installing a package if any of these are set.
* Noting a common fix as unsetting these.
* ((Maybe very) scary option) overriding some of these ourselves.
* Discussion - General info blog post about the the risks of setting these generally (maybe in collaboration with some Linux distro devs and other friends)

Thoughts on these? Other ideas?

cc @kalefranz @msarahan @csoja @isuruf @ocefpaf

**Comment by jakirkham** [comment]
cc @mingwandroid

**Comment by mingwandroid** [comment]
Thanks @jakirkham. For the low level ones:

`LD_LIBRARY_PATH`
Latest conda-build 3 will not create packages with DSOs or exes with DT_RUNPATH in them. So long as everything is rebuilt with this (and the new overlinking detection is set to `error`) then this shouldn't matter to us anymore.

`DYLD_LIBRARY_PATH`
Only applies to people running without SIP which is hardly anyone I expect. I've not seen this cause problems for anyone in a long time.

**Comment by kalefranz** [comment]
All of those are covered in the output of `conda info --all`.  Check it out!

**Comment by kalefranz** [comment]
I'm really hesitant to warn users here.  There are correct ways to use these things.  Or several of them anyway.  I feel like we would annoy and punish the people who are using them correctly.  Seems a little unfair.  But... I also see the value  in these types of warning too.  It's a hard call for me.

**Comment by kalefranz** [comment]
One thing we've done, and you have obviously noticed, is put together an issue template, with some strongly worded warning about ignoring the requested information fields at your own peril.

**Comment by csosborn** [comment]
Note that relying on the `conda` command itself (things like `conda info --all`) may help with general awareness, but not with specific troubleshooting. If `PYTHONHOME` is set there is no guarantee that `conda` will work in the first place. I found my way to this issue because I set `PYTHONHOME` to point to my currently-active conda environment (i.e. `export PYTHONHOME=/miniconda3/envs/test2/`) in hopes of getting a pybind11-based application to work. Lo and behold, my application _did_ start finding the libraries it needed, but `conda` itself began failing.

**Comment by LeslieGerman** [comment]
Not only some warnings when `conda info -a` is executed would be good, but some more detailed documentation. I'm actually confused how to deal with `PYTHONHOME` related problems.
I.e. I installed a python version in a conda env, and tried to run a CPython (native) application in that environment, however it does not run, since `PYTHONHOME`  was not set...

Why conda env did not set `PYTHONHOME` to the python package I installed in that env ?
